aiida/backends/tests/tools/data/orbital/test_orbitals.py

```diff
@@ -0,0 +1,172 @@
+        self.assertAlmostEqual(orbital.get_orbital_dict()['position'][0], -1.)
+        self.assertAlmostEqual(orbital.get_orbital_dict()['position'][1], -2.)
+        self.assertAlmostEqual(orbital.get_orbital_dict()['position'][2], -3.)
+        self.assertAlmostEqual(orbital.get_orbital_dict()['angular_momentum'], 1)
+        self.assertAlmostEqual(orbital.get_orbital_dict()['magnetic_number'], 0)
+        self.assertAlmostEqual(orbital.get_orbital_dict()['radial_nodes'], 2)
+
+    def test_validation_for_fields(self):
+        """Verify that the values are properly validated"""
+        RealhydrogenOrbital = OrbitalFactory('realhydrogen')  # pylint: disable=invalid-name
+
+        with self.assertRaises(ValidationError) as exc:
+            _ = RealhydrogenOrbital(**{
+                'position': (-1, -2, -3),
+                'angular_momentum': 100,
+                'magnetic_number': 0,
+                'radial_nodes': 2
+            })
+        self.assertIn("angular_momentum", str(exc.exception))
+
+        with self.assertRaises(ValidationError) as exc:
+            _ = RealhydrogenOrbital(**{
+                'position': (-1, -2, -3),
+                'angular_momentum': 1,
+                'magnetic_number': 3,
+                'radial_nodes': 2
+            })
+        self.assertIn("magnetic number must be in the range", str(exc.exception))
+
+        with self.assertRaises(ValidationError) as exc:
+            _ = RealhydrogenOrbital(**{
+                'position': (-1, -2, -3),
+                'angular_momentum': 1,
+                'magnetic_number': 0,
+                'radial_nodes': 100
+            })
+        self.assertIn("radial_nodes", str(exc.exception))
+
+    def test_optional_fields(self):
+        """
+        Testing (some of) the optional parameters to check that the functionality works
+        (they are indeed optional but accepted if specified, they are validated, ...)
+        """
+        RealhydrogenOrbital = OrbitalFactory('realhydrogen')  # pylint: disable=invalid-name
+
+        orbital = RealhydrogenOrbital(**{
+            'position': (-1, -2, -3),
+            'angular_momentum': 1,
+            'magnetic_number': 0,
+            'radial_nodes': 2
+        })
+        # Check that the optional value is there and has its default value
+        self.assertEqual(orbital.get_orbital_dict()['spin'], 0)
+        self.assertEqual(orbital.get_orbital_dict()['diffusivity'], None)
+
+        orbital = RealhydrogenOrbital(
+            **{
+                'position': (-1, -2, -3),
+                'angular_momentum': 1,
+                'magnetic_number': 0,
+                'radial_nodes': 2,
+                'spin': 1,
+                'diffusivity': 3.1
+            })
+        self.assertEqual(orbital.get_orbital_dict()['spin'], 1)
+        self.assertEqual(orbital.get_orbital_dict()['diffusivity'], 3.1)
+
+        with self.assertRaises(ValidationError) as exc:
+            _ = RealhydrogenOrbital(
```

Comment:
No need to assign to placeholder if you are just catching exception from constructor